#### Importing Packages 

In [289]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import datetime as dt

import warnings



# Supress warnings
warnings.filterwarnings('ignore')

#### Importing Data

In [290]:
walmart_sales = pd.read_csv('Data\Walmart_Store_sales.csv')

walmart_sales.head()

Store        Date  Weekly_Sales  Holiday_Flag  Temperature  Fuel_Price  \
0      1  05-02-2010    1643690.90             0        42.31       2.572   
1      1  12-02-2010    1641957.44             1        38.51       2.548   
2      1  19-02-2010    1611968.17             0        39.93       2.514   
3      1  26-02-2010    1409727.59             0        46.63       2.561   
4      1  05-03-2010    1554806.68             0        46.50       2.625   

          CPI  Unemployment  
0  211.096358         8.106  
1  211.242170         8.106  
2  211.289143         8.106  
3  211.319643         8.106  
4  211.350143         8.106

The data has been successfully loaded.

**The next step is ***Data Overview and Cleaning***, it is always a good idea to check the data types, missing values and the size of the data.**

In [291]:
walmart_sales.info()    

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6435 entries, 0 to 6434
Data columns (total 8 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Store         6435 non-null   int64  
 1   Date          6435 non-null   object 
 2   Weekly_Sales  6435 non-null   float64
 3   Holiday_Flag  6435 non-null   int64  
 4   Temperature   6435 non-null   float64
 5   Fuel_Price    6435 non-null   float64
 6   CPI           6435 non-null   float64
 7   Unemployment  6435 non-null   float64
dtypes: float64(5), int64(2), object(1)
memory usage: 402.3+ KB


All columns in the dataset contain 6435 non-null values, indicating no apparent missing values. This is a positive sign for data completeness. However, missing values could still exist if encoded in a non-standard format (e.g., placeholders like "N/A" or -1), which will be investigated further during exploratory analysis.

**Converting Date column to datetime datatype**

In [292]:
walmart_sales['Date'] = pd.to_datetime(walmart_sales['Date'], format='%d-%m-%Y')


#### Data Analysis

**Descriptive Statistics**

In [293]:
walmart_sales.describe()    

Store                 Date  Weekly_Sales  Holiday_Flag  \
count  6435.000000                 6435  6.435000e+03   6435.000000   
mean     23.000000  2011-06-17 00:00:00  1.046965e+06      0.069930   
min       1.000000  2010-02-05 00:00:00  2.099862e+05      0.000000   
25%      12.000000  2010-10-08 00:00:00  5.533501e+05      0.000000   
50%      23.000000  2011-06-17 00:00:00  9.607460e+05      0.000000   
75%      34.000000  2012-02-24 00:00:00  1.420159e+06      0.000000   
max      45.000000  2012-10-26 00:00:00  3.818686e+06      1.000000   
std      12.988182                  NaN  5.643666e+05      0.255049   

       Temperature   Fuel_Price          CPI  Unemployment  
count  6435.000000  6435.000000  6435.000000   6435.000000  
mean     60.663782     3.358607   171.578394      7.999151  
min      -2.060000     2.472000   126.064000      3.879000  
25%      47.460000     2.933000   131.735000      6.891000  
50%      62.670000     3.445000   182.616521      7.874000  
75%      74.940000     3.735000   212.743293      8.622000  
max     100.140000     4.468000   227.232807     14.313000  
std      18.444933     0.459020    39.356712      1.875885

**Store Analysis**

In [294]:
store_weekly_sales=walmart_sales.groupby('Store')[['Weekly_Sales']].sum().sort_values(by='Weekly_Sales', ascending=True)

print(f'The first 5 stores with` the lowest sales are:\n')
display(store_weekly_sales[:5])

print(f'The first 5 stores with the highest sales are:\n')
display(store_weekly_sales[-5:])

The first 5 stores with` the lowest sales are:



Weekly_Sales
Store              
33      37160221.96
44      43293087.84
5       45475688.90
36      53412214.97
38      55159626.42

The first 5 stores with the highest sales are:



Weekly_Sales
Store              
2      2.753824e+08
13     2.865177e+08
14     2.889999e+08
4      2.995440e+08
20     3.013978e+08

In [295]:
print(f"The store with maximum sales is Store {store_weekly_sales['Weekly_Sales'].idxmax()} with Sales of {store_weekly_sales['Weekly_Sales'].max():,.2f}\n")

print(f"The store with minimum sales is Store {store_weekly_sales['Weekly_Sales'].idxmin()} with Sales of {store_weekly_sales['Weekly_Sales'].min():,.2f}")


The store with maximum sales is Store 20 with Sales of 301,397,792.46

The store with minimum sales is Store 33 with Sales of 37,160,221.96


In [296]:
store_sales_stats = walmart_sales.groupby('Store')['Weekly_Sales'].agg(['mean', 'std'])

store_sales_stats['cv'] = store_sales_stats['std'] / store_sales_stats['mean']





In [297]:
print(f"The store with maximum sales standard deviations is Store {store_sales_stats['std'].idxmax()} with standard deviations of {store_sales_stats['std'].max():,.2f}\n")

print(f"The store with minimum standard  deviations sales is Store {store_sales_stats['std'].idxmin()} with standard  deviations of {store_sales_stats['std'].min():,.2f}\n")

print(f"Store {store_sales_stats['cv'].idxmax()} has the highest coefficient of variation: {store_sales_stats['cv'].max():.5f}")

The store with maximum sales standard deviations is Store 14 with standard deviations of 317,569.95

The store with minimum standard  deviations sales is Store 37 with standard  deviations of 21,837.46

Store 35 has the highest coefficient of variation: 0.22968


In [298]:
quarterly_sales= walmart_sales.groupby([pd.Grouper(key='Date', freq='QE'),'Store']).agg({'Weekly_Sales': 'sum'}).reset_index()    

In [ ]:

quarterly_sales_q2_and_q3_2012= quarterly_sales[(quarterly_sales['Date'].dt.year == 2012) & (quarterly_sales['Date'].dt.month > 3) & (quarterly_sales['Date'].dt.month <= 9)] 

In [300]:
quarterly_sales_q2_and_q3_2012

Date  Store  Weekly_Sales
405 2012-06-30      1   20978760.12
406 2012-06-30      2   25083604.88
407 2012-06-30      3    5620316.49
408 2012-06-30      4   28454363.67
409 2012-06-30      5    4466363.69
..         ...    ...           ...
490 2012-09-30     41   18093844.01
491 2012-09-30     42    7296759.34
492 2012-09-30     43    8000572.16
493 2012-09-30     44    4411251.16
494 2012-09-30     45    9581268.38

[90 rows x 3 columns]

In [ ]:
qsales_pivot = quarterly_sales_q2_and_q3_2012.pivot(index='Store', columns='Date', values='Weekly_Sales')


qsales_pivot.rename(columns={qsales_pivot.columns[0]: 'Q2_2012', qsales_pivot.columns[1]: 'Q3_2012'}, inplace=True)
qsales_pivot.head() 


Date       Q2_2012      Q3_2012
Store                          
1      20978760.12  20253947.78
2      25083604.88  24303354.86
3       5620316.49   5298005.47
4      28454363.67  27796792.46
5       4466363.69   4163790.99

In [307]:
qsales_pivot['Growth_Rate'] = ((qsales_pivot['Q3_2012'] - qsales_pivot['Q2_2012']) / qsales_pivot['Q2_2012'] * 100).round(2)

qsales_pivot.sort_values(by='Growth_Rate', ascending=False, inplace=True)
qsales_pivot.head(5)

Date       Q2_2012      Q3_2012  Growth_Rate
Store                                       
7       7290859.27   8262787.39        13.33
16      6564335.98   7121541.64         8.49
35     10838313.00  11322421.12         4.47
26     13155335.57  13675691.91         3.96
39     20214128.46  20715116.23         2.48

Store 7 has a good quartely growth of 13.33%  for quater 3 in 2012

In [308]:
qsales_pivot.max()

Date
Q2_2012        28454363.67
Q3_2012        27796792.46
Growth_Rate          13.33
dtype: float64